File: clients/rs/src/features/wrap_type_ints.rs
use std::{error::Error};
use polywrap_client::{core::{uri::Uri, invoker::Invoker}, client::PolywrapClient};
use polywrap_client_default_config::SystemClientConfig;
use polywrap_msgpack_serde;
use serde::{Deserialize, Serialize};
use serde_json::Value;

use crate::input::expect_object;

#[derive(Serialize, Deserialize)]
struct Args {
  first: i32,
  second: i32
}

#[derive(Serialize, Deserialize)]
struct InputObj {
  method: String,
  args: Args,
}

pub fn run_test_case(input_obj: &Value) -> Result<(), Box<dyn Error>> {
  let input_obj = expect_object::<InputObj>(&input_obj)?;
  let method = input_obj.method;
  let args = input_obj.args;

  let binding = std::env::current_dir()?.join("../../wraps");
  let root = binding.to_str().unwrap();
  let uri: Uri = format!("fs/{root}/numbers-type/implementations/as").try_into().unwrap();

  let client: PolywrapClient = PolywrapClient::new(SystemClientConfig::default().into());

  println!("Invoking {method}");

  // TODO: int size?
  let result = client.invoke_raw(
    &uri,
    &method,
    Some(&polywrap_msgpack_serde::to_vec(&args)?),
    None,
    None
  );

  match result {
    Ok(result) => {
      let result = polywrap_msgpack_serde::from_slice::<i32>(&result)?;
      println!("Result: {result:?}");
      println!("Success!");
    },
    Err(err) => panic!("{err}"),
  }
  
  Ok(())
}

File: clients/py/src/features/wrap_type_ints.py
from typing import Any, Dict, TypedDict
from pydantic import BaseModel, Field
from polywrap_client import PolywrapClient
from polywrap_client_config_builder import PolywrapClientConfigBuilder
from polywrap_uri_resolvers import SimpleFileReader, FsUriResolver
from polywrap_core import Uri
from pathlib import Path


class Args(TypedDict):
    first: int
    second: int


class TestCaseInput(BaseModel):
    method: str
    args: Args


def run_test_case(input: Any) -> None:
    input_obj = TestCaseInput.parse_obj(input)

    root = Path(__file__).parent.parent.parent.parent.parent / "wraps"
    uri = Uri.from_str(f"fs/{root}/numbers-type/implementations/as")

    config = (
        PolywrapClientConfigBuilder()
        .add_resolver(FsUriResolver(SimpleFileReader()))
        .build()
    )

    client = PolywrapClient(config)

    print(f"Invoking {input_obj.method}")

    response = client.invoke(
        uri=uri,
        method=input_obj.method,
        args=input_obj.args,
    )

    if not response:
        raise Exception(f"Error: {response}")

    print("Result:", response)
    print("Success!")
